types/derived.gen.go

```diff
@@ -0,0 +1,2067 @@
+func deriveDeepCopy_51(dst, src *PlacementPreferences) {
+	dst.Spread = src.Spread
+	if src.Extensions != nil {
+		dst.Extensions = make(map[string]any, len(src.Extensions))
+		src.Extensions.DeepCopy(dst.Extensions)
+	} else {
+		dst.Extensions = nil
+	}
+}
+
+// deriveDeepCopy_52 recursively copies the contents of src into dst.
+func deriveDeepCopy_52(dst, src *IPAMPool) {
+	dst.Subnet = src.Subnet
+	dst.Gateway = src.Gateway
+	dst.IPRange = src.IPRange
+	if src.AuxiliaryAddresses != nil {
+		dst.AuxiliaryAddresses = make(map[string]string, len(src.AuxiliaryAddresses))
+		deriveDeepCopy_4(dst.AuxiliaryAddresses, src.AuxiliaryAddresses)
+	} else {
+		dst.AuxiliaryAddresses = nil
+	}
+	if src.Extensions != nil {
+		dst.Extensions = make(map[string]any, len(src.Extensions))
+		src.Extensions.DeepCopy(dst.Extensions)
+	} else {
+		dst.Extensions = nil
+	}
+}
+
+// deriveDeepCopy_53 recursively copies the contents of src into dst.
+func deriveDeepCopy_53(dst, src *DeviceRequest) {
+	if src.Capabilities == nil {
+		dst.Capabilities = nil
+	} else {
+		if dst.Capabilities != nil {
+			if len(src.Capabilities) > len(dst.Capabilities) {
+				if cap(dst.Capabilities) >= len(src.Capabilities) {
+					dst.Capabilities = (dst.Capabilities)[:len(src.Capabilities)]
+				} else {
+					dst.Capabilities = make([]string, len(src.Capabilities))
+				}
+			} else if len(src.Capabilities) < len(dst.Capabilities) {
+				dst.Capabilities = (dst.Capabilities)[:len(src.Capabilities)]
+			}
+		} else {
+			dst.Capabilities = make([]string, len(src.Capabilities))
+		}
+		copy(dst.Capabilities, src.Capabilities)
+	}
+	dst.Driver = src.Driver
+	dst.Count = src.Count
+	if src.IDs == nil {
+		dst.IDs = nil
+	} else {
+		if dst.IDs != nil {
+			if len(src.IDs) > len(dst.IDs) {
+				if cap(dst.IDs) >= len(src.IDs) {
+					dst.IDs = (dst.IDs)[:len(src.IDs)]
+				} else {
+					dst.IDs = make([]string, len(src.IDs))
+				}
+			} else if len(src.IDs) < len(dst.IDs) {
+				dst.IDs = (dst.IDs)[:len(src.IDs)]
+			}
+		} else {
+			dst.IDs = make([]string, len(src.IDs))
+		}
+		copy(dst.IDs, src.IDs)
+	}
+}
+
+// deriveDeepCopy_54 recursively copies the contents of src into dst.
+func deriveDeepCopy_54(dst, src *GenericResource) {
+	if src.DiscreteResourceSpec == nil {
+		dst.DiscreteResourceSpec = nil
+	} else {
+		dst.DiscreteResourceSpec = new(DiscreteGenericResource)
+		deriveDeepCopy_55(dst.DiscreteResourceSpec, src.DiscreteResourceSpec)
+	}
+	if src.Extensions != nil {
+		dst.Extensions = make(map[string]any, len(src.Extensions))
+		src.Extensions.DeepCopy(dst.Extensions)
+	} else {
+		dst.Extensions = nil
+	}
+}
+
+// deriveDeepCopy_55 recursively copies the contents of src into dst.
+func deriveDeepCopy_55(dst, src *DiscreteGenericResource) {
+	dst.Kind = src.Kind
+	dst.Value = src.Value
+	if src.Extensions != nil {
+		dst.Extensions = make(map[string]any, len(src.Extensions))
+		src.Extensions.DeepCopy(dst.Extensions)
+	} else {
+		dst.Extensions = nil
+	}
+}
```

Comment:
damn

Reply:
I'm happy I didn't had to write this by hand :)